components/centraldashboard/public/ajax_test_helper.js

```diff
@@ -64,20 +64,21 @@ export function mockIronAjax(component, response, respondWithError = false) {
             response,
             respondWithError,
         );
-        if (respondWithError) {
-            component._setLastError(eventPayload.detail);
-        } else {
+        if (!respondWithError) {
             component._setLastResponse(response);
         }
         component.dispatchEvent(
             new CustomEvent(finalEvent, eventPayload)
         );
         const resp = component.lastResponse || component.lastError;
-
         // So that code can await when using this dynamically
         const completes = new Promise((res, rej) =>
             (respondWithError ? rej : res)(resp)
         );
+        completes.catch((err)=> {
+            component._setLastError(eventPayload.detail);
```

Comment:
If you're going to respond with error after fulfillment – which I expect is to better represent what actually happens in the framework – then do it for both success and error cases to avoid weird timing bugs that could arise.

I would also recommend, in this case, not resolving the promise until a following event loop tick.